src/providers/reversing-distances/reversing-distances.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+import { TestCategory } from '@dvsa/mes-test-schema/categories/common/test-category';
+// TODO: Import CAT C types
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
+import { ReversingLengths } from './reversing-lengths.model';
+
+// TODO: refactor CAT BE to use this provider
+@Injectable()
+export class ReversingDistancesProvider {
+
+  public getDistanceLength(data: Object, category: TestCategory): ReversingLengths {
+    switch (category) {
+      case TestCategory.C:
+        return this.getDistanceLengthCat(data, 3.5);
+      case TestCategory.CE:
+        return this.getDistanceLengthCat(data, 4);
+      case TestCategory.C1:
+        return this.getDistanceLengthCat(data, 3.5);
+      case TestCategory.C1E:
+        return this.getDistanceLengthCat(data, 4);
+      default:
+        return { startDistance: 52.5, middleDistance: 30 };
+    }
+  }
+
+  public getDistanceWidth(data: Object, category: TestCategory): number {
+    switch (category) {
+      case TestCategory.C:
+        return this.getDistanceWidthCat(data);
+      case TestCategory.CE:
+        return this.getDistanceWidthCat(data);
+      case TestCategory.C1:
+        return this.getDistanceWidthCat(data);
+      case TestCategory.C1E:
+        return this.getDistanceWidthCat(data);
+      default:
+        return 3;
+    }
+  }
+
+  // TODO: Use CAT C types
+  private getDistanceWidthCat(data: CatBEUniqueTypes.VehicleDetails): number {
```

Comment:
should it be `getDistanceWidthCatC`?

Reply:
No, the thinking was that the same method can be used for all categories, depending on how the types pan out. Happy to rename to CatC for now though if you prefer?

Reply:
Ok makes sense